src/code.jl

```diff
@@ -146,11 +146,13 @@ function function_to_expr(op::typeof(^), O, st)
             return toexpr(Term(inv, Any[ex]), st)
         else
             args = Any[Term(inv, Any[ex]), -args[2]]
-            op = get(st.rewrites, :nanmath, false) ? op : NaNMath.pow
+            op = get(st.rewrites, :nanmath, false) || args[2] isa Integer ? op : NaNMath.pow
             return toexpr(Term(op, args), st)
         end
     end
-    get(st.rewrites, :nanmath, false) === true || return nothing
+    if !get(st.rewrites, :nanmath, false) || args[2] isa Integer
```

Comment:
A bit safer:
```suggestion
    if get(st.rewrites, :nanmath, false) !== true || args[2] isa Integer
```
Additionally, I guess it would be more efficient to move this to the beginning of the function body?